tornjak-frontend/src/components/agent-list.tsx

```diff
@@ -15,28 +15,45 @@ import {
   agentworkloadSelectorInfoFunc,
   clusterTypeInfoFunc,
 } from 'redux/actions';
+import { RootState } from 'redux/reducers';
+import {
+  AgentsListType, AgentsWorkLoadAttestorInfoType, ServerInfoType,
+} from 'redux/actions/types'
+//import PropTypes from "prop-types";
+
+type AgentListProp = {
+  clusterTypeInfoFunc: (globalClusterTypeInfo: string[]) => void,
+  selectorInfoFunc: (globalSelectorInfo: typeof selectors) => void,
+  workloadSelectorInfoFunc: (globalWorkloadSelectorInfo: typeof workloadSelectors) => void,
+  agentsListUpdateFunc: (globalAgentsList: AgentsListType[]) => void,
+  tornjakMessageFunc: (globalErrorMessage: string) => void,
+  agentworkloadSelectorInfoFunc: (globalAgentsWorkLoadAttestorInfo: AgentsWorkLoadAttestorInfoType[]) => void,
+  tornjakServerInfoUpdateFunc: (globalTornjakServerInfo: Object) => void,
+  serverInfoUpdateFunc: (globalServerInfo: ServerInfoType) => void,
+  globalServerSelected: string,
+  globalErrorMessage: string,
+  globalTornjakServerInfo: Object,
+  agent: Object,
+  globalAgentsList: AgentsListType[] | undefined,
```

Comment:
If no agents are present, the globalAgents list can also be undefined since nothing will be dispatched. Especially during the mounting of components.